sparkmagic/sparkmagic/magics/sparkmagicsbase.py

```diff
@@ -40,6 +41,12 @@ def __init__(self, shell, data=None, spark_events=None):
     def execute_spark(self, cell, output_var, samplemethod, maxrows, samplefraction, session_name, coerce):
         (success, out) = self.spark_controller.run_command(Command(cell), session_name)
         if not success:
+            if conf.spark_statement_errors_are_fatal():
+                if conf.shutdown_session_on_spark_statement_errors():
+                    self.spark_controller.cleanup()
+
+                raise SparkStatementException(out)
+
             self.ipython_display.send_error(out)
```

Comment:
It's possible you'll always want to out out this above any exception raising.

Reply:
Again, I agree with you, I was following how the base code here throws before display: https://github.com/jupyter-incubator/sparkmagic/blob/b9413e590d565390c517b016ba7bc436655f9a79/sparkmagic/sparkmagic/livyclientlib/command.py#L40

Reply:
This occurs at: https://github.com/jupyter-incubator/sparkmagic/blob/b9413e590d565390c517b016ba7bc436655f9a79/sparkmagic/sparkmagic/magics/sparkmagicsbase.py#L41 above my added code, above the `send_error`.